packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -7,19 +11,27 @@ import {
 } from '@metamask/controller-utils';
 import type {
   NetworkClientId,
-  NetworkController,
-  NetworkState,
+  NetworkControllerGetNetworkClientByIdAction,
+  NetworkControllerGetStateAction,
+  NetworkControllerStateChangeEvent,
 } from '@metamask/network-controller';
-import { StaticIntervalPollingControllerV1 } from '@metamask/polling-controller';
-import type { PreferencesState } from '@metamask/preferences-controller';
+import { StaticIntervalPollingController } from '@metamask/polling-controller';
+import type {
+  PreferencesControllerGetStateAction,
+  PreferencesControllerStateChangeEvent,
+} from '@metamask/preferences-controller';
 import { createDeferredPromise, type Hex } from '@metamask/utils';
 import { isEqual } from 'lodash';
 
 import { reduceInBatchesSerially, TOKEN_PRICES_BATCH_SIZE } from './assetsUtil';
 import { fetchExchangeRate as fetchNativeCurrencyExchangeRate } from './crypto-compare-service';
 import type { AbstractTokenPricesService } from './token-prices-service/abstract-token-prices-service';
 import { ZERO_ADDRESS } from './token-prices-service/codefi-v2';
-import type { TokensControllerState } from './TokensController';
+import type {
+  TokensControllerGetStateAction,
+  TokensControllerStateChangeEvent,
+  TokensControllerState,
+} from './TokensController';
 
 /**
  * @type Token
```

Comment:
I have noticed many types missing properties definition, I don't know if can force this with  a linting rule

Reply:
As a starting point:

```ts
 * @property aggregators - An array containing the token's aggregators
 * @property hasBalanceError - 'true' if getting the balance of an ERC-20 token resulted in an error
 * @property isErc721 - 'true' if the token is an NFT
 * @property name - Name of the token
```

Reply:
@cryptodev-2s I'm sure there are jsdoc eslint rules that are supposed to do that. I'll write up a ticket for enabling them.

Reply:
That said, it looks like we're moving away from jsdoc comments for types, since annotating each property individually gives the user better access to the comments, especially when hovering on the property.

(e.g. see https://github.com/MetaMask/core/pull/4286/files#diff-ea0aa37793484e278bd59970b7a6ce8380b75b723b34612081d402fb66261cfdR152-R196, and hover over any search result of "networkConfiguration." (`networkConfiguration\.` for regex) in the same file.

For functions, objects etc., we might still want to enforce jsdoc lint rules, so I'll still take a look into configuring this.

Reply:
Adding an ESLint rule to check for JSDoc'ing properties would be great! I did some initial research into this a while back and didn't find any existing rules but maybe someone has come up with something by now.